src/FwdState.cc

```diff
@@ -392,6 +392,9 @@ FwdState::Start(const Comm::ConnectionPointer &clientConn, StoreEntry *entry, Ht
         internalStart(clientConn, request, entry, al);
         return;
     }
+    // Requests for our cache manager must never be forwarded elsewhere. If
+    // reach this method, the internalStart() call above must satisfy them.
+    Assure(!ForThisCacheManager(*request));
```

Comment:
Together, this and internalStart() checks cover miss routing decisions. I think we should also cover hit routing. I suggest addressing clientReplyContext::identifyStoreObject() XXX with something like the following code:

```C++
    HttpRequest *r = http->request;

    if (r->flags.noCache) {
        // no-cache requests skip Store lookups
        identifyFoundObject(nullptr, "no-cache");
    } else {
        // cache manager requests ought to have flags.noCache (handled above)
        Assure(!ForThisCacheManager(*r));

        const auto e = storeGetPublicByRequest(r);
        identifyFoundObject(e, storeLookupString(bool(e)));
    }
```

```